### PR TITLE
separate & refactor rich text

### PR DESCRIPTION
**Pull request by sundevour**
does d1 have objective descriptions? does it have rich ones? how was this working up until this point?

is there something i can do, given defs, to check if they are d2? there's gotta be a correcter way to do RichDestinyText lines 3/4/103

### CODE REVIEW

**Comment by delphiactual** [comment]
No D1 does not have any rich text

**Comment by delphiactual** [comment]
276438067 - small blocker
3792840449 - medium blocker
2031240843 - large blocker

File: src/app/dim-ui/RichDestinyText.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    (r) => r.substring === textSegment || r.unicode === textSegment
+  );
+  return (
+    (replacement && <img src={replacement.icon} title={textSegment} key={textSegment} />) || (
+      <span key={textSegment}>{textSegment}</span>
+    )
+  );
+};
+
+export default function RichDestinyText({
```
Comment (bhollis) [inline src/app/dim-ui/RichDestinyText.tsx]: Mind writing a bit of description for this?

File: src/app/dim-ui/RichDestinyText.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  text,
+  defs
+}: {
+  text: string;
+  defs?: D2ManifestDefinitions | D1ManifestDefinitions;
```
Comment (bhollis) [inline src/app/dim-ui/RichDestinyText.tsx]: D1 won't have any of these - they always used just text for all descriptions. We could add a `destinyVersion` property to the manifest definitions object that you could check if you wanted.

**Comment by sundevour** [comment]
image

**Comment by sundevour** [comment]
https://github.com/DestinyItemManager/DIM/blob/071d09737daa0cd92540cada1430462550ad99d9/src/app/destiny2/definitions.ts

@bhollis wtf or perfectly reasonable

**Comment by bhollis** [comment]
Seems totally reasonable

**Pull request merged**